pilot/pkg/xds/filters/filters.go

```diff
@@ -113,7 +113,9 @@ var (
 	TLSInspector = &listener.ListenerFilter{
 		Name: wellknown.TLSInspector,
 		ConfigType: &listener.ListenerFilter_TypedConfig{
-			TypedConfig: protoconv.MessageToAny(&tlsinspector.TlsInspector{}),
+			TypedConfig: protoconv.MessageToAny(&tlsinspector.TlsInspector{
+				InitialReadBufferSize: &wrapperspb.UInt32Value{Value: 512}, // Default is 64KB.
```

Comment:
Did we get a chance to check the typical Istio size? In linkerd its 512 but they use ECC certs by default which should be smaller

Reply:
Oh nice so we need the ClientHello which is ~170, cert comes later and we don't need it

Reply:
post-quantum crypto (tldr.fail) seems to increase the cert size beyond MTU, so a large-ish MTU (8k) is probably a better bound?

Reply:
Would it break for such cases?

Reply:
We don't ues post-quantum crypto in our TLS handshakes so I think it makes sense to optimize around our actual use cases, and niche cases can expand the buffer.

Reply:
@howardjohn We are using the same boringssl library as Chrome - we certainly have code to enable post-quantum, and it'll eventually be enabled in Envoy. Given the lack of configurability in TLSv1.3, we may not be able to opt-out.

@ramaraochavali No, it's not going to break with a short buffer, but you'd see excessive CPU usage possibly.

Reply:
I think for those type of use cases, people can use envoy filter to change this value?